channel/handlers.go

```diff
@@ -2979,3 +2993,167 @@ func (server Server) handleChatEvent(conn mnet.Server, reader mpacket.Reader) {
 		log.Println("Unknown chat event type:", op)
 	}
 }
+
+func (d *player) removeAllCooldowns() {
+	if d == nil || d.skills == nil {
+		return
+	}
+	for id, ps := range d.skills {
+		ps.Cooldown = 0
+		ps.TimeLastUsed = 0
+		d.updateSkill(ps)
+		_ = id
+	}
+}
+
+func (server *Server) getAffectedPartyMembers(p *party, affectedMask byte) []*player {
+	if p == nil {
+		return nil
+	}
+
+	members := make([]*player, 0, constant.MaxPartySize)
+	for _, member := range p.players { // ensure party has a method to enumerate member IDs
+		members = append(members, member)
+	}
+	_ = affectedMask // apply bit filtering here when member order is known
+	return members
+}
+
+func (server *Server) playerSpecialSkill(conn mnet.Client, reader mpacket.Reader) {
+	// Minimal, safe implementation to keep packet stream in sync and apply basic validations/costs.
+	plr, err := server.players.getFromConn(conn)
+	if err != nil {
+		return
+	}
+
+	// Dead players cannot cast
+	if plr.hp == 0 {
+		plr.send(packetPlayerNoChange())
+		return
+	}
+
+	// The packet layout for special skills generally starts with [skillID int32][skillLevel byte]
+	skillID := reader.ReadInt32()
+	skillLevel := reader.ReadByte()
+
+	// Validate the player owns the skill and level does not exceed learned level
+	ps, ok := plr.skills[skillID]
+	if !ok || skillLevel == 0 || skillLevel > ps.Level {
+		// Possible hack/desync; drop request
+		plr.send(packetPlayerNoChange())
+		return
+	}
+
+	_ = reader.ReadByte()       // party flags
+	delay := reader.ReadInt16() // delay
+
+	readMobListAndDelay := func() {
+		count := int(reader.ReadByte())
+		for i := 0; i < count; i++ {
+			_ = reader.ReadInt32() // mob spawn id
+		}
+		_ = reader.ReadInt16() // delay
+	}
+	readXY := func() {
+		_ = reader.ReadInt16()
+		_ = reader.ReadInt16()
+	}
+
+	applied := false
+
+	switch skill.Skill(skillID) {
+	// Party buffs handled earlier remain unchanged...
+	case skill.Haste, skill.BanditHaste, skill.Bless, skill.IronWill, skill.Rage,
+		skill.Meditation, skill.ILMeditation, skill.MesoUp, skill.HolySymbol, skill.HyperBody:
+		if plr.party == nil {
+			plr.addBuff(skillID, skillLevel, delay)
+			plr.send(packetPlayerSkillAnimThirdParty(plr.id, false, true, skillID, skillLevel))
```

Comment:
This should be sent to the instance as other players should see when you have applied a skill buff